packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -592,7 +644,10 @@ class _PopupMenuRouteLayout extends SingleChildLayoutDelegate {
 
   @override
   bool shouldRelayout(_PopupMenuRouteLayout oldDelegate) {
-    return position != oldDelegate.position;
+    return position != oldDelegate.position
+        || selectedItemIndex != oldDelegate.selectedItemIndex
+        || textDirection != oldDelegate.textDirection
+        || listEquals(itemSizes, oldDelegate.itemSizes);
```

Comment:
Shouldn't it be this?

```suggestion
        || !listEquals(itemSizes, oldDelegate.itemSizes);
```

Also, is there a test that changes the length of the list (it doesn't seem so)?